pennylane/transforms/optimization/commute_controlled.py

```diff
@@ -13,31 +13,162 @@
 # limitations under the License.
 """Transforms for pushing commuting gates through targets/control qubits."""
 
-from itertools import chain
-
 from pennylane import apply
 from pennylane.wires import Wires
 from pennylane.transforms import qfunc_transform
 
 from .optimization_utils import find_next_gate
 
-# Gates that commute among each particular basis
-commuting_gates = {
-    "X": ["PauliX", "RX", "SX"],
-    "Y": ["PauliY", "RY"],
-    "Z": ["PauliZ", "RZ", "PhaseShift", "S", "T"],
-}
 
-movable_gates = list(chain(*commuting_gates.values()))
+def _commute_controlled_right(op_list):
+    """Push commuting single qubit gates to the right of controlled gates.
+
+    Args:
+        op_list (list[Operation]): The initial list of operations.
+
+    Returns:
+        list[Operation]: The modified list of operations with all single-qubit
+        gates as far right as possible.
+    """
+    # We will go through the list backwards; whenever we find a single-qubit
+    # gate, we will extract it and push it through 2-qubit gates as far as
+    # possible to the right.
+    current_location = len(op_list) - 1
+
+    while current_location >= 0:
+
+        current_gate = op_list[current_location]
+
+        # We are looking only at the gates that can be pushed through
+        # controls/targets; these are single-qubit gates with the basis
+        # property specified.
+        if current_gate.basis is None or len(current_gate.wires) != 1:
+            current_location -= 1
+            continue
+
+        # Find the next gate that contains an overlapping wire
+        next_gate_idx = find_next_gate(current_gate.wires, op_list[current_location + 1 :])
+
+        new_location = current_location
+
+        # Loop as long as a valid next gate exists
+        while next_gate_idx is not None:
+            next_gate = op_list[new_location + next_gate_idx + 1]
+
+            # Only go ahead if information is available
+            if next_gate.basis is None:
+                break
```

Comment:
I guess this never happens in the test suite?

Reply:
Coverage has been added for this and the instances in `_commute_controlled_left` :+1: